crates/rnote-ui/src/workspacebrowser/filerow/actions/trash.rs

```diff
@@ -1,30 +1,19 @@
 // Imports
-use crate::workspacebrowser::RnFileRow;
 use crate::RnAppWindow;
-use gettextrs::gettext;
-use gtk4::{gio, glib, glib::clone, prelude::FileExt};
-use tracing::debug;
+use crate::{dialogs, workspacebrowser::RnFileRow};
+use gtk4::{gio, glib, glib::clone};
 
 /// Create a new `trash` action.
 pub(crate) fn trash(filerow: &RnFileRow, appwindow: &RnAppWindow) -> gio::SimpleAction {
     let action = gio::SimpleAction::new("trash-file", None);
-    action.connect_activate(
-        clone!(@weak filerow, @weak appwindow => move |_action_trash_file, _| {
-            let Some(current_file) = filerow.current_file() else {
-                return;
-            };
-            current_file.trash_async(
-                glib::source::Priority::DEFAULT,
-                None::<&gio::Cancellable>,
-                clone!(@weak filerow, @strong current_file => move |res| {
-                if let Err(e) = res {
-                    appwindow.overlays().dispatch_toast_error(&gettext("Trashing file failed"));
-                    debug!("Trash filerow file `{current_file:?}` failed , Err: {e:?}");
-                    return;
-                }
-                filerow.set_current_file(None);
-            }));
-        }),
-    );
+    action.connect_activate(clone!(@weak appwindow, @weak filerow => move |_, _| {
+        let Some(current_file) = filerow.current_file() else {
+            return;
+        };
+        glib::spawn_future_local(clone!(@weak appwindow, @strong current_file => async move {
+            dialogs::dialog_trash_file(&appwindow, &current_file).await;
+        }));
+        filerow.set_current_file(None);
```

Comment:
Could it be that this is the cause that the dialog only appears once? I think `dialog_trash_file` would need to give back if the file was actually deleted and only based on that set the file to `None`.

Reply:
Actually, if I just leave this out it seems to work fine. Do you see any problems with that? Worst case I'd just overwrite the value on the next action, no? The reason I moved all the logic into the dialogue in the first place was that it seemed kinda cumbersome to communicate from the dialogue back what choice was made.

Also, going to close this PR in favour of a new one since the merge conflicts are too time-consuming to properly fix for me.

Reply:
You're right, it probably is fine to leave it out entirely